src/services/csv_service.py

```diff
@@ -21,15 +21,44 @@ def get_covid_states_csv(state=None):
     return state_dict
 
 
-def get_covid_counties_csv():
+def get_covid_by_county_csv(state, county):
     '''
     Get COVID-19 cases by county
     '''
-    pass
+    county_df = pd.read_csv(f'src/resources/{Constants.COUNTIES_FILENAME}')
+    covid_data = county_df[(county_df['state'] == state) & (county_df['county'] == county)].to_dict("records")
+    
+    if covid_data == []:
+        return Error.NOT_FOUND
+
+    return covid_data
 
 
 def get_total_covid_cases_csv():
     '''
     Get COVID-19 cases in the US
     '''
     pass
+
+
+def get_state_county_info():
+    '''
+    Get all counties for every state 
+    '''
+
+    # get a list of all states
+    county_df = pd.read_csv(f'src/resources/{Constants.COUNTIES_FILENAME}')
+    states = list(county_df['state'].unique())
+
+    # get a list of the counties in a state
+    counties_by_state = dict()
+    for state in states:
+        counties = list(county_df[county_df['state'] == state]['county'].unique())
+        if counties == ['Unknown']:
+            # happens for US territories that don't have counties
+            counties_by_state[state] = []
```

Comment:
instead of returning an empty list, if its a US territory without counties (or DC), let's return a 404 here. there are no counties for this territory.

Reply:
I don't think a 404 makes sense. It's not getting the counties for a specific state, it's creating a dictionary where the key is the state/territory name and the value is an array of county names in that state/territory.  This will be called when the page initially loads so it knows what states and territories to list as options, as well as what counties to list for a state when it is selected.

Reply:
oh yes, it would help if i could read